### PR TITLE
Wip/rebased/dev

### PR DESCRIPTION
**Pull request by bewest**
This is an updated version of the dev branch made to be able to merge with master.

https://github.com/nightscout/minimed-connect-to-nightscout/pull/1

Apologies if the git trail is confusing, we had some diverging patches and branches at one point.  Our best efforts have been taken to preserve the history.

### CODE REVIEW

File: carelink.js [hunk 1]
```diff
@@ -24,21 +33,27 @@ var CARELINK_SECURITY_URL = 'https://' + carelinkServerAddress + '/patient/j_sec
...
 
 var Client = exports.Client = function (options) {
+    let requestCount = 0;
+
     if (!(this instanceof Client)) {
         return new Client(arguments[0]);
     }
 
     axiosCookieJarSupport(axios);
-    var cookieJar = new tough.CookieJar();
```
Comment (mrinnetmaki) [inline carelink.js:38]: Agree the cookie jar is now initiated two times on consecutive rows. However, if you remove the var `cookieJar` here, it is undefined in row 281...

**Review by mrinnetmaki — COMMENTED** [review]
Good stuff @bewest!

I had another (or perhaps supplementary) approach for a safe multi-tenant implementation (regarding your comment https://github.com/nightscout/minimed-connect-to-nightscout/issues/11#issuecomment-739419508).

I believe we should use an axios instance per client. See #18.

Just pointing that out in case it would trigger any thoughts on the logins and cookie handling.

**Comment by bewest** [comment]
Thanks, @mrinnetmaki I was able to fix that error as well as rebase your idea over here: https://github.com/nightscout/minimed-connect-to-nightscout/pull/19

Something is not quite right for the US test though, even though it is fine in this branch.

**Comment by bewest** [comment]
I don't know if this works in Argentina, but this seems to work in the US and in EU.  Thanks for the extra patches, @mrinnetmaki.

File: carelink.js (hunk 1 above)
Comment (bewest) [reply carelink.js:38]: I believe this was due to a mistake on my end while resolving conflicts and should be fixed.

**Pull request merged**